core/src/main/java/com/netflix/conductor/core/execution/tasks/Join.java

```diff
@@ -119,10 +119,13 @@ public boolean execute(
     @Override
     public Optional<Long> getEvaluationOffset(TaskModel taskModel, long defaultOffset) {
         int index = taskModel.getPollCount() > 0 ? taskModel.getPollCount() - 1 : 0;
-        if (index == 0) {
+        // Assuming SystemTaskWorker 50ms pollInterval this will cause the join to be evaluated
+        // continuously during the first 5 seconds and the FORK/JOIN will end without any delay.
+        if (index <= 100) {
```

Comment:
To reduce the delay of the JOIN task when all forked tasks complete, poll continuously for reasonable amount of time (useful for fork/joins that are meant to finish in less than N seconds).

Then gradually increase the offset which introduces the delay in the JOIN task evaluation - (might end up changing the exponential backoff and will make the N configurable)

Reply:
Or we make JOIN sync again (?)